compiler/src/iree/compiler/Pipelines/Pipelines.cpp

```diff
@@ -107,6 +108,8 @@ void buildIREEPrecompileTransformPassPipeline(
     case InputDialectOptions::Type::none:
       break;
     case InputDialectOptions::Type::auto_detect:
+      // The VHLO -> StableHLO passes will be a Nop if there is no VHLO present.
+      ::mlir::stablehlo::createStablehloDeserializePipeline(passManager);
       // Run the auto pipeline that chooses from plugins using module contents.
       passManager.addPass(
           createAutoInputConversionPipelinePass(hooks.pipelineExtensions));
```

Comment:
This pipeline is generic. Anything specific to an individual input dialect must go in that dialect's input plugin.

Reply:
fixed.